### PR TITLE
Don't error when passing an empty array to a polymorphic association (rails 4.1.10 regression)

### PR DESCRIPTION
**Pull request by huoxito**
We're seeing an issue similar to #19087 on spree builds running rails 4.1.10. Applying that patch to 4-1-stable fixes the issue.

sorry we didn't catch this earlier guys ..

### CODE REVIEW

**Comment by rafaelfranca** [comment]
@huoxito COMMON Piauí! :trollface: 

Thanks, I'll take a look. :smile:

**Comment by huoxito** [comment]
lol thanks a lot @rafaelfranca !

**Comment by JDutil** [comment]
@rafaelfranca I think you've only been asking Spree to test release candidates for the latest version of rails, but we probably could have caught this at the same time as the 4.2.1.rc regression if we had tested 4.1.10.rc feel free to ping us to test the older supported rails versions as well.

**Comment by rafaelfranca** [comment]
@JDutil, yeah, my fault for not including both releases on the issue. I usually do this.

File: activerecord/test/cases/relation/where_test.rb [hunk 1]
```diff
@@ -96,6 +96,18 @@ def test_polymorphic_nested_array_where
       assert_equal expected.to_sql, actual.to_sql
     end
 
+    def test_polymorphic_empty_array_where
+      treasure = Treasure.new
+      treasure.id = 1
+      hidden = HiddenTreasure.new
+      hidden.id = 2
+
+      expected = PriceEstimate.where("1=0")
```
Comment (kassio) [inline activerecord/test/cases/relation/where_test.rb]: I guess you could use `PriceEstimate.none` here. :smiley:

Comment (huoxito) [inline activerecord/test/cases/relation/where_test.rb]: true :) `none` is new to me, but I just cherry-picked the commit, not sure it's worth it to update, amend and trigger another ci run (lazy monday)

**Comment by sgrif** [comment]
Thanks for catching this.

**Comment by arthurnn** [comment]
> @huoxito COMMON Piauí! :trollface:
LOL

**Comment by JDutil** [comment]
@sgrif no worries but if we catch a latest rails version but thats also backported to older branches that we don't mention assume thats bad too :) No worries though!

**Pull request merged**